Riot/Managers/EncryptionKeyManager/EncryptionKeyManager.swift

```diff
@@ -0,0 +1,122 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import UIKit
+import KeychainAccess
+import MatrixKit
+import CommonCrypto
+
+@objcMembers
+class EncryptionKeyManager: NSObject, MXKeyProviderDelegate {
+    static let shared = EncryptionKeyManager()
+    
+    private static let keychainService: String = BuildSettings.baseBundleIdentifier + ".encryption-manager-service"
+    private static let contactsIv: KeyValueStoreKey = "contactsIv"
+    private static let contactsAesKey: KeyValueStoreKey = "contactsKey"
+    private static let accountIv: KeyValueStoreKey = "acountIv"
+    private static let accountAesKey: KeyValueStoreKey = "acountKey"
+    private static let realmCryptoKey: KeyValueStoreKey = "realmCryptoKey"
```

Comment:
You can let these variable as is. Just to mention that you can also group some variable sharing the same purpose like this:
```
private enum StoreKeys {
static let contactsIv: KeyValueStoreKey
static let contactsAes: KeyValueStoreKey
static let realmCrypto: KeyValueStoreKey
...
}

```

Reply:
Good idea but I'm not sure we need enum in this case.

Reply:
It can be a `struct` with just constants. Using an `enum` in this sample is just to indicate that we cannot instantiate it.